filters/auth/jwt_validation.go

```diff
@@ -0,0 +1,257 @@
+
+	issuerURL := sargs[0]
+
+	cfg, err := getOpenIDConfig(issuerURL)
+	if err != nil {
+		return nil, err
+	}
+
+	var ac *authClient
+	var ok bool
+	if ac, ok = issuerAuthClient[issuerURL]; !ok {
+		ac, err = newAuthClient(cfg.JwksURI, tokenInfoSpanName, s.options.Timeout, s.options.MaxIdleConns, s.options.Tracer)
+		if err != nil {
+			return nil, filters.ErrInvalidFilterParameters
+		}
+		issuerAuthClient[issuerURL] = ac
+	}
+
+	f := &jwtValidationFilter{
+		authClient: ac,
+	}
+	f.claims = strings.Split(sargs[1], " ")
+	if !all(f.claims, cfg.ClaimsSupported) {
+		return nil, fmt.Errorf("%v: %s, supported Claims: %v", errUnsupportedClaimSpecified, strings.Join(f.claims, ","), cfg.ClaimsSupported)
+	}
+
+	// inject additional headers from the access token for upstream applications
+	if len(sargs) > 2 && sargs[2] != "" {
+		f.upstreamHeaders = make(map[string]string)
+
+		for _, header := range strings.Split(sargs[2], " ") {
+			sl := strings.SplitN(header, ":", 2)
+			if len(sl) != 2 || sl[0] == "" || sl[1] == "" {
+				return nil, fmt.Errorf("%w: malformatted filter for upstream headers %s", filters.ErrInvalidFilterParameters, sl)
+			}
+			f.upstreamHeaders[sl[0]] = sl[1]
+		}
+		log.Debugf("Upstream Headers: %v", f.upstreamHeaders)
+	}
+
+	return f, nil
+}
+
+// String prints nicely the jwtValidationFilter configuration based on the
+// configuration and check used.
+func (f *jwtValidationFilter) String() string {
+	return fmt.Sprintf("%s(%s)", JwtValidationName, strings.Join(f.claims, ","))
+}
+
+func (f *jwtValidationFilter) validateAnyClaims(token jwt.Token) bool {
+	for _, wantedClaim := range f.claims {
+		if claims, ok := token.Claims.(jwt.MapClaims); ok {
+			if _, ok2 := claims[wantedClaim]; ok2 {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (f *jwtValidationFilter) Request(ctx filters.FilterContext) {
+	r := ctx.Request()
+
+	var info jwt.Token
+	infoTemp, ok := ctx.StateBag()[oidcClaimsCacheKey]
+	if !ok {
+		token, ok := getToken(r)
+		if !ok || token == "" {
+			unauthorized(ctx, "", missingToken, f.authClient.url.Hostname(), "")
+			return
+		}
+
+		body, err := f.authClient.getTokeninfo("", ctx)
+		if err != nil {
+			log.Errorf("Error while getting jwt keys: %v.", err)
+
+			unauthorized(ctx, "", "jwt public keys", f.authClient.url.Hostname(), "")
+			return
+		}
+		//var body map[string]interface{}
+		//json.NewDecoder(resp.Body).Decode(&body)
+		rsakeys = make(map[string]*rsa.PublicKey)
+		if body["keys"] != nil {
+			for _, bodykey := range body["keys"].([]interface{}) {
+				key := bodykey.(map[string]interface{})
+				kid := key["kid"].(string)
+				rsakey := new(rsa.PublicKey)
+				number, _ := base64.RawURLEncoding.DecodeString(key["n"].(string))
+				rsakey.N = new(big.Int).SetBytes(number)
+				rsakey.E = 65537
+				rsakeys[kid] = rsakey
+			}
+		} else {
+			log.Error("Not able to get public keys")
+			unauthorized(ctx, "", "Not able to get public keys", f.authClient.url.Hostname(), "")
+			return
+		}
+
```

Comment:
Please see https://github.com/zalando/skipper/pull/1811/files#r672359596 and check that you do not need https://github.com/zalando/skipper/pull/1811/files#diff-7632350376ed0aebec65031a9b5191d7ea917bfe9fbc8b0f87e0c7b865572304R146-R171

Starting from `body, err := f.authClient.getTokeninfo("", ctx)` until the comment, you do not need.

Reply:
This is clear in general. Will try to implement the change.